aws/resource_aws_apigatewayv2_model_test.go

```diff
@@ -0,0 +1,283 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAWSAPIGatewayV2Model_basic(t *testing.T) {
+	var apiId string
+	var v apigatewayv2.GetModelOutput
+	resourceName := "aws_apigatewayv2_model.test"
+	rName := strings.ReplaceAll(acctest.RandomWithPrefix("tf-acc-test"), "-", "")
+
+	schema := `
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "ExampleModel",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
+    }
+  }
+}
+`
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAPIGatewayV2ModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAPIGatewayV2ModelConfig_basic(rName, schema),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayV2ModelExists(resourceName, &apiId, &v),
+					resource.TestCheckResourceAttr(resourceName, "content_type", "application/json"),
+					resource.TestCheckResourceAttr(resourceName, "description", ""),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "schema", schema),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateIdFunc: testAccAWSAPIGatewayV2ModelImportStateIdFunc(resourceName),
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSAPIGatewayV2Model_disappears(t *testing.T) {
+	var apiId string
+	var v apigatewayv2.GetModelOutput
+	resourceName := "aws_apigatewayv2_model.test"
+	rName := strings.ReplaceAll(acctest.RandomWithPrefix("tf-acc-test"), "-", "")
+
+	schema := `
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "ExampleModel",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
+    }
+  }
+}
+`
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSAPIGatewayV2ModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSAPIGatewayV2ModelConfig_basic(rName, schema),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSAPIGatewayV2ModelExists(resourceName, &apiId, &v),
+					testAccCheckAWSAPIGatewayV2ModelDisappears(&apiId, &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSAPIGatewayV2Model_AllAttributes(t *testing.T) {
+	var apiId string
+	var v apigatewayv2.GetModelOutput
+	resourceName := "aws_apigatewayv2_model.test"
+	rName := strings.ReplaceAll(acctest.RandomWithPrefix("tf-acc-test"), "-", "")
+
+	schema1 := `
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "ExampleModel1",
+  "type": "object",
+  "properties": {
+    "id": {
+      "type": "string"
+    }
+  }
+}
+`
+	schema2 := `
```

Comment:
For readability, please update the formatting on `schema2` to match other JSON strings